src/core/server/http/router/router.ts

```diff
@@ -153,39 +155,28 @@ export class Router {
     routeSchemas: RouteSchemas<P, Q, B> | undefined,
     request: Request,
     responseToolkit: ResponseToolkit,
-    handler: RequestHandler<P, Q, B>
+    handler: RequestHandler<P, Q, B>,
+    log: Logger
   ) {
     let kibanaRequest: KibanaRequest<TypeOf<P>, TypeOf<Q>, TypeOf<B>>;
-
+    const hapiResponseAdapter = new HapiResponseAdapter(responseToolkit);
     try {
       kibanaRequest = KibanaRequest.from(request, routeSchemas);
     } catch (e) {
-      // TODO Handle failed validation
-      return responseToolkit.response({ error: e.message }).code(400);
+      return hapiResponseAdapter.toBadRequest(e.message);
     }
 
     try {
       const kibanaResponse = await handler(kibanaRequest, responseFactory);
-
-      let payload = null;
-      if (kibanaResponse.payload instanceof Error) {
-        // TODO Design an error format
-        payload = { error: kibanaResponse.payload.message };
-      } else if (kibanaResponse.payload !== undefined) {
-        payload = kibanaResponse.payload;
-      }
-
-      return responseToolkit.response(payload).code(kibanaResponse.status);
+      return hapiResponseAdapter.handle(kibanaResponse);
     } catch (e) {
-      // TODO Handle `KibanaResponseError`
-
-      // Otherwise we default to something along the lines of
-      return responseToolkit.response({ error: e.message }).code(500);
+      log.error(e);
```

Comment:
should we log `Boom` errors differently, prepending with a message that HTTP server no longer supports this way of rejecting a request?